langchain/src/vectorstores/xata.ts

```diff
@@ -0,0 +1,102 @@
+import { BaseClient } from "@xata.io/client";
+import { VectorStore } from "./base.js";
+import { Embeddings } from "../embeddings/base.js";
+import { Document } from "../document.js";
+
+export interface XataClientArgs<XataClient> {
+  readonly client: XataClient;
+  readonly table: string;
+}
+
+type XataFilter = object;
+
+export class XataVectorSearch<
+  XataClient extends BaseClient
+> extends VectorStore {
+  private readonly client: XataClient;
+
+  private readonly table: string;
+
+  _vectorstoreType(): string {
+    return "xata";
+  }
+
+  constructor(embeddings: Embeddings, args: XataClientArgs<XataClient>) {
+    super(embeddings, args);
+
+    this.client = args.client;
+    this.table = args.table;
+  }
+
+  async addDocuments(documents: Document[], options?: { ids?: string[] }) {
+    const texts = documents.map(({ pageContent }) => pageContent);
+    return this.addVectors(
+      await this.embeddings.embedDocuments(texts),
+      documents,
+      options
+    );
+  }
+
+  async addVectors(
+    vectors: number[][],
+    documents: Document[],
+    options?: { ids?: string[] }
+  ) {
+    const rows = vectors
+      .map((embedding, idx) => ({
+        content: documents[idx].pageContent,
+        embedding,
+        ...documents[idx].metadata,
+      }))
+      .map((row, idx) => {
+        if (options?.ids) {
+          return { id: options.ids[idx], ...row };
+        }
+        return row;
+      });
+
+    const res = await this.client.db[this.table].createOrReplace(rows);
```

Comment:
Going to assume this is sanitized? Documents can have arbitrary metadata

Reply:
Sanitized against SQL injection or similar? Yes. Or do you mean other type of sanitization?

Reply:
Yeah injection. Good!